src/notification.c

```diff
@@ -438,45 +498,31 @@ void notification_init(notification *n)
                 g_free(n->msg);
                 n->msg = buffer;
         }
+}
 
-        n->dup_count = 0;
-
-        /* urgency > URG_CRIT -> array out of range */
-        if (n->urgency < URG_MIN)
-                n->urgency = URG_LOW;
-        if (n->urgency > URG_MAX)
-                n->urgency = URG_CRIT;
-
-        if (!n->color_strings[ColFG]) {
-                n->color_strings[ColFG] = xctx.color_strings[ColFG][n->urgency];
-        }
-
-        if (!n->color_strings[ColBG]) {
-                n->color_strings[ColBG] = xctx.color_strings[ColBG][n->urgency];
-        }
-
-        if (!n->color_strings[ColFrame]) {
-                n->color_strings[ColFrame] = xctx.color_strings[ColFrame][n->urgency];
-        }
-
-        n->timeout =
-            n->timeout < 0 ? settings.timeouts[n->urgency] : n->timeout;
-        n->start = 0;
-
-        n->timestamp = g_get_monotonic_time();
-
-        n->redisplayed = false;
+static void notification_extract_urls(notification *n)
+{
+        g_clear_pointer(&n->urls, g_free);
 
-        n->first_render = true;
+        char *concat = g_strconcat(n->summary, " ", n->body, NULL);
+        // remove <a href=... to not confuse extract_urls
+        char *concat_plain = markup_transform(g_strdup(concat), MARKUP_FULL);
+        char *urls_plain;
```

Comment:
What's the point of this variable? You're assigning and freeing it.

Reply:
It **is** actually used. But you're objection has a good reason. The way this variable gets assigned is shit.